tests/vmi_vsock_test.go

```diff
@@ -0,0 +1,160 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"encoding/xml"
+	"strconv"
+
+	expect "github.com/google/goexpect"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+)
+
+var _ = Describe("[Serial][sig-compute]VSOCK", func() {
+	var virtClient kubecli.KubevirtClient
+	var err error
+
+	BeforeEach(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("VM creation", func() {
+		It("should expose a VSOCK device", func() {
+			checks.SkipTestIfNoFeatureGate(virtconfig.VSOCKGate)
+			By("Creating a VMI with VSOCK enabled")
+			vmi := tests.NewRandomFedoraVMI()
+			vmi.Spec.Domain.Devices.AutoattachVSOCK = pointer.Bool(true)
+			vmi = tests.RunVMIAndExpectLaunch(vmi, 60)
+			Expect(vmi.Status.VSOCKCID).NotTo(BeNil())
+
+			By("creating valid libvirt domain")
+
+			domain, err := tests.GetRunningVirtualMachineInstanceDomainXML(virtClient, vmi)
+			Expect(err).ToNot(HaveOccurred())
+			domSpec := &api.DomainSpec{}
+			Expect(xml.Unmarshal([]byte(domain), domSpec)).To(Succeed())
+			Expect(domSpec.Devices.VSOCK.CID.Auto).To(Equal("no"))
+			cid, err := strconv.ParseUint(domSpec.Devices.VSOCK.CID.Address, 10, 32)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cid).To(BeEquivalentTo(*vmi.Status.VSOCKCID))
+
+			By("Logging in as root")
+			err = console.LoginToFedora(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Ensuring a vsock device is present")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "ls /dev/vsock-vhost\n"},
+				&expect.BExp{R: "/dev/vsock-vhost"},
+			}, 300)).To(Succeed(), "Could not find a vsock-vhost device")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "ls /dev/vsock\n"},
+				&expect.BExp{R: "/dev/vsock"},
```

Comment:
In the second part where we add the stream support, we can add an e2e  tests where an actual `hello` or so would be sent back from the guest. For this phase the tests are complete enough from my point of view. :+1: